### DISCUSSION TITLE
Custom coil .ccd

### DISCUSSION THREAD

**Discussion by VictorS-M** [simnibs/simnibs#389] [8. Simulation workflows] [answered]
Hello to everyone.
I'm using a custom coil model in SimNibs with a .ccd file and I want to make a TMS simulation. The file header is:

# Coil1.ccd version 1.0 number of elements; dIdtmax=141.0;x=-300,300;y=-200,200;z=-200,200;resolution=3;stimulator=MagPro X100;brand=MagVenture;dIdtstim=151;coilname=Coil1;

With this header I want SimNibs to do the calculation using the MagPro x100 stimulator but I have some doubts:
- Is this a good way to work with a custom coil model?
- Is the  waveform of the pulse in the .tsv file?
- How can I know the value of dI/dt in order to produce, for example, an electric field of 100 V/m in the cortex?

Thak you so much.

**Comment by Kstoffer** [comment, marked as answer]
The information in the header of the ccd file is not used in the simulations as such, for now it mainly serves as information about dI/dt and the desired FOV and resolution of a .nii.gz transformed version. Also, the waveform is not used in the simulation but intended for use in future versions of simNIBS which may take dynamic properties or neural models into account (but yes it is in the tsv file - this will depend on the inductance of the coil and the properties of the stimulator).
You cannot really know which dI/dt is required before running a simulation with a changed coil, but you can just scale the result afterwards the problem is linear.
Just write if you have additional questions or concerns, best
Kristoffer

**Comment by VictorS-M** [comment]
Thank you very much for your reply.
Now everything is clearer.
